app/Http/Controllers/Worker/SupervisorController.php

```diff
@@ -57,8 +62,42 @@ public function editWorker(Workplace $workplace, Worker $worker) {
 
 	}
 
-	public function updateWorker(UpdateWorkerRequest $request,Workplace $workplace, Worker $worker) {
+	public function updateWorker(UpdateWorkerRequest $request,Workplace $workplace, Shift $worker) {
 		return $request->commit();
 	}
 
+	public function editShift(Workplace $workplace, Shift $shift) {
+		$fields = $shift->fullData->map(function ($value){
+			$value['readonly'] = true;
+			return $value;
+		});
+		$shiftWorkers = $shift->workers->map(function ($worker) use ($shift){
+			$shift = $worker->shifts->find($shift);
```

Comment:
this is also unnecessary, you already have the shift

Reply:
Also this line repeats like 3 times?

Reply:
this doesnt work like this, cuz i want it with the pivot for this worker

Reply:
fixed the reappearnaces